### DISCUSSION TITLE
Nested functions.

### DISCUSSION THREAD

**Discussion by Digital39999** [honojs/hono#3220] [General]
Is there any reason why [status](https://github.com/honojs/hono/blob/1fafc7aec87f1afe68b37506df58a66acd9f29aa/src/context.ts#L505) or such don't return `this` so it allows for easier nesting? I mean I get that its a little useless, but I write that if block so many times that it already annoys me because it can be nicer inline with single change. I didn't want to directly create a PR, so I'm asking here.

Instead of:
```js
if (!condition) {
    <Context>.status(401);
    return <Context>.json({ message: 'You are not authorized.' })
}
```

to just do:
```js
if (!condition) return c.status(401).json({ message: 'You are not authorized.' });
```

**Comment by EdamAme-x** [comment]
In that case, you can use such as following code.

```ts
if (!condition) return c.json({ message: 'You are not authorized.' }, 401);
```

Isn't it?

**Reply by Digital39999** [reply]
I had no idea that exists, I forgot to look at the types as I just switched from express and fastify so I guess I was just used to nesting, but good to know lol, thanks.

**Comment by yusukebe** [comment]
Hi @Digital39999 

Implementing a feature allowing the following syntax is not a bad idea.

```ts
c.status(401).json({ message: 'You are not authorized.' })
```

But, actually, we've never had trouble if `status` does not return a Context instance. As @EdamAme-x said, you can write like that code. So, I think we don't have to implement it now.

**Reply by Digital39999** [reply]
Yeah I agree, my bad for not seeing it before.
